include/alpaka/core/DemangleTypeNames.hpp

```diff
@@ -0,0 +1,18 @@
+/* Copyright 2022 Andrea Bocci, Antonio Di Pilato
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <boost/core/demangle.hpp>
+
+namespace alpaka::core
+{
+    template<typename T>
+    inline const std::string demangle = boost::core::demangle(typeid(T).name());
```

Comment:
```suggestion
    template<typename T>
    inline const std::string demangled = boost::core::demangle(typeid(T).name());
```
I would call it `demangled`, because it is not a function so a verb seems inappropriate.

Reply:
However, we could argue that `f<x>` is just another spelling for a function `f(x)` taking a compile time input (a meta function). So my argument is not a strong one.